go/cmd/vtgateclienttest/services/errors.go

```diff
@@ -5,27 +5,34 @@
 package services
 
 import (
+	"errors"
 	"fmt"
+	"strings"
 
 	"golang.org/x/net/context"
 
+	"github.com/youtube/vitess/go/vt/callerid"
+	"github.com/youtube/vitess/go/vt/vterrors"
 	"github.com/youtube/vitess/go/vt/vtgate/proto"
 	"github.com/youtube/vitess/go/vt/vtgate/vtgateservice"
 
+	mproto "github.com/youtube/vitess/go/mysql/proto"
 	pb "github.com/youtube/vitess/go/vt/proto/topodata"
 	pbg "github.com/youtube/vitess/go/vt/proto/vtgate"
+	"github.com/youtube/vitess/go/vt/proto/vtrpc"
 )
 
 // errorClient implements vtgateservice.VTGateService
 // and returns specific errors. It is meant to test all possible error cases,
 // and make sure all clients handle the errors correctly.
-//
-// So far, we understand:
-// - "return integrity error": Execute* will return an integrity error.
-// - "error": GetSrvKeyspace will return an error.
-//
-// TODO(alainjobart) Add throttling error.
-// TODO(alainjobart) Add all errors the client may care about.
+
+const (
+	// ErrorPrefix is the prefix to send with queries so they go through this service handler.
+	ErrorPrefix = "error://"
+	// PartialErrorPrefix is the prefix to send with queries so the RPC call returns a partial error.
```

Comment:
Because we just love our ATM machines...

I suspect it's because we have to differentiate between RPC protocols and actual RPCs. 

Anyways, fixed.